internal/provider/resource_credential_json_test.go

```diff
@@ -0,0 +1,196 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/credentials"
+	"github.com/hashicorp/boundary/testing/controller"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+const (
+	jsonCredResc = "boundary_credential_json.example"
+	jsonCredName = "bar"
+	jsonCredDesc = "the bar"
+	jsonCredObj  = `jsonencode({
+		password = "password",
+		username = "admin"
+	})`
+	jsonCredObjUpdate = `jsonencode({
+		password = "password",
+		username = "db-admin"
+	})`
+)
+
+func jsonCredResource(name, description, object string) string {
+	return fmt.Sprintf(`
+resource "boundary_credential_store_static" "example" {
+	name  = "static store name"
+	description = "static store description"
+	scope_id = boundary_scope.proj1.id
+	depends_on = [boundary_role.proj1_admin]
+}
+
+resource "boundary_credential_json" "example" {
+	name  = "%s"
+	description = "%s"
+	credential_store_id = boundary_credential_store_static.example.id
+	object = %s
+}`, name, description, object)
+}
+
+func TestAccCredentialJson(t *testing.T) {
+	tc := controller.NewTestController(t, tcConfig...)
+	defer tc.Shutdown()
+	url := tc.ApiAddrs()[0]
+
+	res := jsonCredResource(
+		jsonCredName,
+		jsonCredDesc,
+		jsonCredObj,
+	)
+
+	resUpdate := jsonCredResource(
+		jsonCredName,
```

Comment:
Would be good to update the name and description and validate they both are updated in the update as well.